src/Corporation/Actions.ts

```diff
@@ -356,20 +370,22 @@ export function MakeProduct(
   division.products[product.name] = product;
 }
 
-export function Research(division: IIndustry, researchName: string): void {
+export function Research(division: IIndustry, researchName: string): boolean {
   const researchTree = IndustryResearchTrees[division.type];
   if (researchTree === undefined) throw new Error(`No research tree for industry '${division.type}'`);
   const allResearch = researchTree.getAllNodes();
   if (!allResearch.includes(researchName)) throw new Error(`No research named '${researchName}'`);
   const research = ResearchMap[researchName];
 
-  if (division.sciResearch.qty < research.cost)
-    throw new Error(`You do not have enough Scientific Research for ${research.name}`);
+  if (division.hasResearch(researchName)) return true;
+  if (division.sciResearch.qty < research.cost) return false;
```

Comment:
for example these changes would instead of bubbling up why it failed,  would just return false.

granted there is a generic message to handle it in all the cases i could see so its probably not a big deal, but if there needed to be multiple possible errors it would no longer be possible.

Reply:
though i don't really have  a abetter suggestion so ill approve for now 👍

Reply:
I can change this (or any other of the modified methods) to return string | null instead with the error message if the error should be non-fatal and then write to the tail console instead if you'd prefer, I mostly don't do that because the failure reasons tend to be quite simple and predictable. Let me know if you'd like some changes.

Reply:
lets wait for others to weigh in tbh 👍